File: packages/python-sdk/src/treedb_sdk/adapters/federation.py
from __future__ import annotations

from typing import Any

from .common import json_request
from treedb_sdk.transport import Transport


class FederationAdapter:
    def __init__(self, transport: Transport) -> None:
        self.transport = transport

    def plan(self, body: Any) -> Any:
        return json_request(self.transport, "POST", "/api/v1/federation/query/plan", body)

    def search(self, body: Any) -> Any:
        return json_request(self.transport, "POST", "/api/v1/search", body)

    def query(self, body: Any) -> Any:
        return json_request(self.transport, "POST", "/api/v1/query", body)

    def context_build(self, body: Any) -> Any:
        return json_request(self.transport, "POST", "/api/v1/context/build", body)

    def graph_query(self, body: Any) -> Any:
        return json_request(self.transport, "POST", "/api/v1/graph/query", body)

File: packages/elixir-sdk/lib/tree_db_sdk/adapters/federation.ex
defmodule TreeDbSdk.Federation do
  @moduledoc false
  alias TreeDbSdk.Adapters.Common

  def plan(client, body) do
    Common.json_request(client, :post, "/api/v1/federation/query/plan", body, %{})
  end

  def search(client, body) do
    Common.json_request(client, :post, "/api/v1/search", body, %{})
  end

  def query(client, body) do
    Common.json_request(client, :post, "/api/v1/query", body, %{})
  end

  def context_build(client, body) do
    Common.json_request(client, :post, "/api/v1/context/build", body, %{})
  end

  def graph_query(client, body) do
    Common.json_request(client, :post, "/api/v1/graph/query", body, %{})
  end
end
